README.md

```diff
@@ -50,6 +50,123 @@ The aim of DSF is not just to make *_building_* your components easier, but also
 Not every design system should be built in the same way, but where possible we want to empower design-system authors by sharing  tools that make building design-systems simpler and faster.
 
 # Design Systems Framework Patterns
+
 ## RenderProps
+
+### When to:
+Render props are really useful in scenarios where the component author has opinions to enforce about functionality of the component that has little to do with how the primitive html elements are rendered.
+
+Take the following form example:
+
+```
+<Form>
+  ({ formProps, fieldProps }) => (
+    <form {...formProps}>
+      <input type="text" {...fieldProps}/>
+    </form>
+  )
+</Form>
+```
+
+In this case the <Form> component really only wants to manage state and events around the captured data from the form specified by the consumer. What the form elements are, how they're rendered on the page and in what order is completely irrelevant. In this case renderProps are an excellent pattern, as we are afforded incredible flexibility around which elements get rendered and how, while still providing a clean and simple API for the functionality that we wish to take ownership of.
```

Comment:
This could be a hook, couldn’t it? There are definitely some render prop components that can’t be replaced with hooks but, in most cases, shouldn’t hooks be preferred?

Reply:
@mitchellhamilton Yup! This right now is not actually a recommendation of patterns to use, but an examination of renderProps as a well established pattern (where it is effective and where it is less effective), and the scenarios where components prop might make more sense. 

Imo renderProps and hooks are both different approaches to solving a very similar / same class of problem (and actually would like thoughts on this). That said, will add adding mention of hooks and a hooks example to my TODO

Reply:
> Imo renderProps and hooks are both different approaches to solving a very similar / same class of problem (and actually would like thoughts on this)

I think they certainly have been used to solve the same class of problem but they each give slightly different amounts of control and there are still use cases for render props in a world with hooks. Hooks let you create abstractions that can implement state and side effects but they don't have control over what is rendered. _Most_ render props don't need this control over what's rendered so a lot of render props will likely be replaced with hooks but there are some that need this control. A real world example of this is Emotion's `ClassNames` component, it wouldn't make as much sense as a hook because it needs to be able to have control over what react elements are being returned(to return the style tags during SSR) but also expose part of what should be rendered.

A good way to explain it might be, if a render prop implementation looks like 
```jsx
class MyCoolComponent extends React.Component {
  // ...lifecyles and stuff
  render() {
    return this.props.children({...someStuff})
  }
}
```

But if a component looks more like